src/pcre2_compile.c

```diff
@@ -6139,13 +6142,16 @@ for (;; pptr++)
     else
       {
       *code++ = OP_CLASS;
-      memset(code, 0, 32 * sizeof(uint8_t));
+      memset(code, 0, 32);
       code += 32 / sizeof(PCRE2_UCHAR);
       }
 
     if (firstcuflags == REQ_UNSET) firstcuflags = REQ_NONE;
     zerofirstcu = firstcu;
     zerofirstcuflags = firstcuflags;
+    // XXX revisit why these flags here are different to when we process a
+    // non-empty class - is this a problem? You can have '[^\d\D]' is basically
+    // the same as '[]' so they should surely set the same flags.
```

Comment:
This is a question for reviewers.

The handling for META_CLASS_EMPTY_NOT emits OP_ALLANY, and sets the flags:

>    if (firstcuflags == REQ_UNSET) firstcuflags = REQ_NONE;
>    zerofirstcu = firstcu;
>    zerofirstcuflags = firstcuflags; 

And, the handling for `[\d\D]` also emits OP_ALLANY, but sets the different flags:

>    if (firstcuflags == REQ_UNSET) firstcuflags = REQ_NONE;
>    zerofirstcu = firstcu;
>    zerofirstcuflags = firstcuflags;
>    zeroreqcu = reqcu;
>    zeroreqcuflags = reqcuflags;

Is this a problem? Is there any impact?

Reply:
I haven't spent time to understand these constructs, so this is also magic to me.

Reply:
Perhaps @PhilipHazel could comment on this one specific question.

Reply:
Always happy to hepple. :-)